### PR TITLE
Upgrade dotnet CLI version

### PR DESCRIPTION
**Pull request by dagood**
See https://github.com/dotnet/corefx/pull/6337 for reference. This PR does the same changes, but in buildtools. Because buildtools isn't restored using xplat nuget already in the official build like corefx/wcf, some fixes to project.json files fixing restore compatibility errors are in this PR.

Also includes `www.myget.org` -> `dotnet.myget.org` for dotnet feeds to stop relying on the redirect.

The commit messages may help explain some changes.

/cc @ericstj @weshaggard

### CODE REVIEW

**Comment by ellismg** [comment]
> Also includes www.myget.org -> dotnet.myget.org for dotnet feeds to stop relying on the redirect.

Yay!

**Comment by ericstj** [comment]
LGTM

File: dir.props [hunk 1]
```diff
@@ -98,18 +98,16 @@
   <!-- list of nuget package sources passed to dnu -->
   <ItemGroup>
     <!-- Need to escape double forward slash (%2F) or MSBuild will normalize to one slash on Unix. -->
-    <DnuSourceList Include="https:%2F%2Fwww.myget.org/F/dotnet-core/" />
-    <DnuSourceList Include="https:%2F%2Fwww.myget.org/F/dotnet-corefxtestdata/" />
-    <DnuSourceList Include="https:%2F%2Fwww.myget.org/F/dotnet-buildtools/" />
+    <DnuSourceList Include="https:%2F%2Fdotnet.myget.org/F/dotnet-core/" />
```
Comment (weshaggard) [inline dir.props]: Can you also change this to v3 index.json?

**Comment by joperezr** [comment]
Nice! LGTM

File: dir.props (hunk 1 above)
Comment (dagood) [inline dir.props]: Yep, and I realized I needed to make these source changes to NuGet.Config too.

**Comment by dagood** [comment]
@dotnet-bot test this please

**Comment by weshaggard** [comment]
@chcosta this is going to intersect with your change to reduce the feeds.

**Comment by dagood** [comment]
Holding off on merging this until after @chcosta is done (and I fix that CI error and any more that pop up. I'm fairly sure the current one is just due to merging in non-merge-conflicting changes that introduce compat errors).

File: BuildToolsVersion.txt [hunk 2]
```diff
@@ -1 +1 @@
-1.0.25-prerelease-00158
\ No newline at end of file
+1.0.25-prerelease-00177
```
Comment (jhendrixMSFT) [inline BuildToolsVersion.txt]: Can we bump this to build 00180?  It's needed to pick up some recent changes to support MicroBuild v2.

Comment (dagood) [inline BuildToolsVersion.txt]: Sure. I'll get this PR mergeable now too.

**Pull request merged**